app/models/proxy_config.rb

```diff
@@ -104,7 +104,7 @@ def update_version
 
     # Read the value
     version = config.connection.select_value(config.select(:version)).to_i
-    write_attribute_without_type_cast 'version', version
+    _write_attribute 'version', version
```

Comment:
The previous one was also internal. But the question is good - what is the purpose of using an internal method.

The only purpose I can think of is to avoid attribute being counted as dirty (not persisted). idk if this change retains that. Maybe not.

Because see what method does - updates the version atomically but outside active record, then updates the attribute `version` with the value from database. These operations will mark attribute as dirty. But it shouldn't be marked as such because we took the value from database.

Perhaps this one will need a little more attention. Initially I didn't put so much thought into it. I only saw that the old method was gone and another low-overhead method was used. So good that you asked.

Reply:
> why not just `version += 1`, see comment a little above - atomic updating based on another query

Reply:
To be honest, the method seemed too complex to me to investigate, I just used whatever seemed to be the replacement of the now missing method...

Reply:
This is my suggestion for a change

```suggestion
    version = version
    clear_attribute_changes [:version]
```

Reply:
In fact, I am not sure `clear_attribute_changes` is needed :thinking: 
Seems to be working fine without it.

Reply:
it is an optimization, not something to change the workings.

Try with and without it
```
pc = FactoryBot.create(:proxy_config, proxy: Proxy.take)
pc.version_changed?
```

With it, there is no change reported which is the purpose. Because we have just read the value from database. So it is not dirty.